### PR TITLE
feat(examples): Introduce HTTP Rust server as example

### PR DESCRIPTION
**Pull request by razvand**
Introduce an HTTP Rust server as binary compatibility run. Build server as a PIE application using `Dockerfile`. Then run it with the `base` kernel images from `../../kernels/`.

Add typical files for a bincompat app:

* `Kraftfile`: build / run rules, including pulling the `base` image
* `Dockerfile`: filesystem, including binary and libraries
* `Makefile`: used to generate the root filesystem from the `Dockerfile` rules
* `README.md`: instructions to set up, build and run the application
* `config.yaml`: configuration file to generate scripts to the application
* `http_server.rs`: the Rust HTTP server

`config.yaml` is used to generate run scripts using the `../../utils/bincompat/generate.py` script.

The kernels in `../../kernels` are generated by running the `../../utils/bincompat/base-build-all.sh` script while inside the `../../library/base/` directory.

### CODE REVIEW

File: examples/http-rust1.73/Kraftfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+spec: v0.6
+
+name: http-rust
+
+runtime: index.unikraft.io/unikraft.org/base:latest
+
+rootfs: ./Dockerfile
+
+cmd: ["/server"]
```
Comment (StefanJum) [inline examples/http-rust1.73/Kraftfile:9]:
```suggestion
cmd: ["/http_server"]
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
See the comment, works fine otherwise.

**Review by Mihnea0Firoiu — APPROVED** [review]
Looks good.
Reviewed-by: Mihnea Firoiu <mihneafiroiu0@gmail.com>

**Review by StefanJum — APPROVED** [review]
All good, thanks.

Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request closed without merging**